pkg/network/tcp_transport.go

```diff
@@ -84,3 +78,20 @@ func (t *TCPTransport) Close() {
 func (t *TCPTransport) Proto() string {
 	return "tcp"
 }
+
+// Address implements the Transporter interface.
+func (t *TCPTransport) Address() string {
+	if t.listener == nil {
+		t.addListener()
```

Comment:
What should we do if `t.listener` is not yet set? In case of dynamic port `Listener.Address()` method might return different value each time we create new listener.

Reply:
> What should we do if t.listener is not yet set?

Return error? Return ""? It depends on `Address()` users mostly, what would be easier to handle/more appropriate to get.

> In case of dynamic port Listener.Address() method might return different value each time we create new listener.

That's a feature.

Reply:
Fixed this, now it does not affect `Transporter` and returns "" if `listener` is not set.